plugins/builds/update.js

```diff
@@ -197,7 +195,10 @@ async function isStageDone({ stage, event }) {
     const stageJobBuilds = await event.getBuilds({ params: { jobId: stageJobIds } });
     let stageIsDone = false;
 
-    if (stageJobBuilds && stageJobBuilds.length !== 0) {
+    // Make sure all builds in stage have run successfully in success case
+    if (stageJobBuilds && buildStatus === 'SUCCESS') {
+        stageIsDone = stageJobBuilds.length === stageJobIds.length;
```

Comment:
We should not conclude the stage as completed when the `stageJobBuilds.length` match `stageJobIds.length`. We also need to ensure that all those builds have reached terminal status.

Reply:
I think below logic to update the status of stage build is buggy
https://github.com/screwdriver-cd/screwdriver/pull/3216/files#diff-2d039e9e4eec4a54c7c110debffb3a176c9774982fc9f9d783a5a750defaa2d2R317-R322

When the stage job build is successful, we might be prematurely marking the stage build as `SUCCESS` without checking if other stage job builds are pending execution.